### PR TITLE
test: add e2e test for date picker component 

### PR DESCRIPTION
**Pull request by nikvalor**
Covered date picker component 
#### Please provide a link to the associated issue.
[4346](https://app.zenhub.com/workspaces/sap-ui-framework-5f32933d64202b0012e3572e/issues/sap/fundamental-ngx/4346)
#### Please provide a brief summary of this pull request.
Added e2e test for date picker component 

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: e2e/wdio/platform/tests/date-picker.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    beforeAll(() => {
+        datePickerPage.open();
+        waitForPresent(datePickerPage.inputDatePicker);
+    });
+
+    afterEach(() => {
+        refreshPage();
+        waitForPresent(datePickerPage.inputDatePicker);
+    });
+
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/date-picker.e2e-spec.ts:22]: please add the retires like below:

beforeAll(() => {
        datePickerPage.open();
        waitForPresent(datePickerPage.inputDatePicker);
    }, 1);

    afterEach(() => {
        refreshPage();
        waitForPresent(datePickerPage.inputDatePicker);
    }, 1);

File: e2e/wdio/platform/tests/date-picker.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+} from '../../driver/wdio';
+import { DatePicker } from '../pages/date-picker.po';
+import datePickerTestData from '../fixtures/testData/date-picker';
+
+describe('Date picker suite', function() {
+    const datePickerPage: DatePicker = new DatePicker();
+
+    beforeAll(() => {
+        datePickerPage.open();
+        waitForPresent(datePickerPage.inputDatePicker);
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/date-picker.e2e-spec.ts:15]: I think you could move this waitForPresent to the open method in the PO file; I did it like that for all other test specs.

**Review by Seamoo13 — CHANGES_REQUESTED** [review]

**Review by Seamoo13 — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
@nikvalor please maintain the checkboxes in the PR template. `Mergeable` check will fail if not all checkboxes are checked.

**Review by droshev — APPROVED** [review]

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+export class DatePicker extends BaseComponentPo {
+    url = '/date-picker';
+    root = '#page-content';
+
+    inputDatePicker = 'fd-date-picker input';
+    buttonDatePicker = 'fd-date-picker button';
+
+    activeButtonDatePicker = '[ng-reflect-is-disabled="false"] + span button';
+    activeInputDatePicker = '[ng-reflect-is-disabled="false"][aria-label="Date input"]';
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:12]: try to avoid binding to the name

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    calendarExpanded = '.fd-popover__popper fd-calendar';
+    calendarYearsSection = '.fd-calendar__content--years';
+
+    currentYear = '[aria-label="Select year"]';
+    currentDay = '//*[contains(@class, "fd-calendar__item--current") or contains(@class, "is-active")]';
+
+    buttonGerman = 'button[ng-reflect-label="German"]'
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:20]: try to avoid binding to the name

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    buttonBulgarian = 'button[ng-reflect-label="Bulgarian"]';
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:21]: same

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    buttonSelectYear = '[aria-label="Select year"]';
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:22]: same

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    buttonSelectMonth = '[aria-label="Select month"]';
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:23]: same

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    inputDatePicker = 'fd-date-picker input';
+    buttonDatePicker = 'fd-date-picker button';
+
+    activeButtonDatePicker = '[ng-reflect-is-disabled="false"] + span button';
+    activeInputDatePicker = '[ng-reflect-is-disabled="false"][aria-label="Date input"]';
+
+    calendarExpanded = '.fd-popover__popper fd-calendar';
+    calendarYearsSection = '.fd-calendar__content--years';
+
+    currentYear = '[aria-label="Select year"]';
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:17]: try to avoid binding to the name

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    currentDay = '//*[contains(@class, "fd-calendar__item--current") or contains(@class, "is-active")]';
+
+    buttonGerman = 'button[ng-reflect-label="German"]'
+    buttonBulgarian = 'button[ng-reflect-label="Bulgarian"]';
+    buttonSelectYear = '[aria-label="Select year"]';
+    buttonSelectMonth = '[aria-label="Select month"]';
+
+    buttonFirstRangeYear = '(//td[contains(@id,"fd-aggregated-year")]/child::span)[1]'
+    buttonFirstYear = '(//td[contains(@id,"year")]/child::span)[1]'
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:26]: check formatting  (;)

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    buttonFirstMonth = '(//td[contains(@id,"month")]/child::span)[1]'
+
+    filterCaledarValue = (name: string) => {
+        return `//td[contains(@id,"fd-${name}")]`;
+    };
+
+    datePickerInputByIndex = (index: string) => {
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:33]: look like it is auto-generated  id's
maybe better use fd-popover tag with index?

File: e2e/wdio/platform/pages/date-picker.po.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    buttonGerman = 'button[ng-reflect-label="German"]'
+    buttonBulgarian = 'button[ng-reflect-label="Bulgarian"]';
+    buttonSelectYear = '[aria-label="Select year"]';
+    buttonSelectMonth = '[aria-label="Select month"]';
+
+    buttonFirstRangeYear = '(//td[contains(@id,"fd-aggregated-year")]/child::span)[1]'
+    buttonFirstYear = '(//td[contains(@id,"year")]/child::span)[1]'
+    buttonFirstMonth = '(//td[contains(@id,"month")]/child::span)[1]'
+
+    filterCaledarValue = (name: string) => {
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/date-picker.po.ts:29]: (name: string): string

File: e2e/wdio/platform/tests/date-picker.e2e-spec.ts [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+            click(datePickerPage.activeButtonDatePicker, i);
+            click(datePickerPage.buttonSelectMonth);
+            waitForElDisplayed(datePickerPage.filterCaledarValue('month'));
+            click(datePickerPage.buttonFirstMonth)
+            waitForElDisplayed(datePickerPage.filterCaledarValue('day'));
+            click(datePickerPage.activeButtonDatePicker, i);
+        }
+    });
+
+    it('verify user is cannot select mutlitple dates', () => {
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/tests/date-picker.e2e-spec.ts:233]: typo

File: e2e/wdio/platform/pages/date-picker.po.ts (hunk 11 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/pages/date-picker.po.ts:29]: Caledar typo

**Review by AntonOlkhovskyi — CHANGES_REQUESTED** [review]

**Review by AntonOlkhovskyi — APPROVED** [review]

**Pull request merged**